### PR TITLE
Stream cutscene audio

### PR DESCRIPTION
**Pull request by donnierussellii**
Convert and stream chunks of cutscene audio instead of converting it all at once.

Audiologs stream the same way. Commits done.

### CODE REVIEW

**Comment by Interrupt** [comment]
It's kind of weird having the SDL Mixer callback in the XMI file in this case where it could be streaming either cutscene audio or music - maybe it should get moved to a more generalized place?

**Comment by Interrupt** [comment]
Found a function that could help us out a lot, SDL_mixer has a `Mix_HookMusic` callback which is a callback stream used exclusively for music: https://www.libsdl.org/projects/SDL_mixer/docs/SDL_mixer_60.html

I think we could use that for music streaming with the added benefit of being able to control the music volume simply with `Mix_VolumeMusic`, and also free us up to use the other mixer callback exclusively for Cutscene / Audiolog streaming.

**Comment by Interrupt** [comment]
Thanks for being patient with this PR, ended up with some bad food poisoning this last weekend so I haven't been able to really dig into this yet.

**Comment by donnierussellii** [comment]
Hope you're feeling better.

This seems to work well. I listened to most of the audiologs and barks throughout a playthrough with no problems. I put up a video on my youtube channel of levels 1 and 2. I love how fast the audio starts now.

File: src/GameSrc/audiolog.c [hunk 1]
```diff
@@ -37,315 +34,235 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+extern char which_lang;
+
+
+
 extern void MacTuneUpdateVolume(void);
 
-#define ALOG_MUSIC_DUCK  0.7
 
-//-------------------------------------------------------------
-//  Sorry excuse for a function.
```
Comment (Interrupt) [inline src/GameSrc/audiolog.c]: This is a stylistic choice, but it's fun to see some of these original comments from the code. In the future, I'd err towards keeping things like this over throwing them away just for posterity's sake.

**Pull request merged**